### DISCUSSION TITLE
[Miscellaneous] Providing Sequencer Feed Messages

### DISCUSSION THREAD

**Discussion by gloomydumber** [ethereum-optimism/developers#623] [Miscellaneous]
### Did you check the documentation and specs?

- [X] Yes

### Was there anything unclear or missing?

whether there are any plans to provide a WebSocket server that streams sequencer feed messages in real-time on Optimism.

### Did you check for duplicate questions?

- [X] Yes

### Please select the type of request

- [ ] Bug Report
- [X] Feature Request

### Feature Description

Provide a WebSocket server that streams sequencer feed messages in real-time on the Optimism network.

### Purpose and Benefits

The main purpose of this feature is to enhance transparency and security within the Optimism network. By providing a WebSocket server that streams sequencer feed messages in real-time, it allows anyone to monitor the sequencer’s actions. This ensures that any attempt by the sequencer to manipulate transactions or alter their order can be detected immediately, increasing trust in the protocol. Real-time monitoring of sequencer feeds also strengthens the decentralized nature of the network by making it easier for third parties to verify the integrity of the transaction processing.

### Relevant Context or Examples

Arbitrum provides real-time sequencer feeds via WebSocket,

**Comment by imtipi** [comment]
i don't know do optimism has that tool rightnow, but there's a third party website can help you decode the batch
https://www.blocknative.com/l2-decoding

**Comment by Chomtana** [comment]
There shouldn't be a websocket version but you can implement yourself by developing a server to listening to libp2p gossipsub topic in the same way op-node does and expose the Websocket endpoint.

The libp2p related code is in this folder https://github.com/ethereum-optimism/optimism/tree/develop/op-node/p2p
